lib/src/trailer.rs

```diff
@@ -0,0 +1,264 @@
+            let key = caps[1].trim().to_string();
+            multiline_value.push(caps.get(2).unwrap().as_str());
+            let value = multiline_value.iter().rev().join(" ");
+            multiline_value.clear();
+            trailer.push(Trailer { key, value });
+        } else if line.trim().is_empty() {
+            // end of the trailer
+            found_blank = true;
+            break;
+        } else {
+            // a non trailer in the trailer paragraph
+            // the line is ignored, as well as the multiline value that may
+            // have previously been accumulated
+            multiline_value.clear();
+        }
+    }
+    // reverse the insert order, since we parsed the trailer in reverse
+    trailer.reverse();
+    (trailer, found_blank)
+}
+
+#[cfg(test)]
+mod tests {
+    use pretty_assertions::assert_eq;
+
+    use super::*;
+
+    #[test]
+    fn test_simple_trailers() {
+        let body = r#"chore: update itertools to version 0.14.0
+
+Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed
+do eiusmod tempor incididunt ut labore et dolore magna aliqua.
+
+Co-authored-by: Alice <alice@example.com>
+Co-authored-by: Bob <bob@example.com>
+Reviewed-by: Charlie <charlie@example.com>
+Change-Id: I1234567890abcdef1234567890abcdef12345678"#;
+
+        let trailer = parse_description_trailers(body);
+        assert_eq!(trailer.len(), 4);
+
+        assert_eq!(trailer[0].key, "Co-authored-by");
+        assert_eq!(trailer[0].value, "Alice <alice@example.com>");
+
+        assert_eq!(trailer[1].key, "Co-authored-by");
+        assert_eq!(trailer[1].value, "Bob <bob@example.com>");
+
+        assert_eq!(trailer[2].key, "Reviewed-by");
+        assert_eq!(trailer[2].value, "Charlie <charlie@example.com>");
+
+        assert_eq!(trailer[3].key, "Change-Id");
+        assert_eq!(
+            trailer[3].value,
+            "I1234567890abcdef1234567890abcdef12345678"
+        );
+    }
+
+    #[test]
+    fn test_trailers_with_colon_in_body() {
+        let body = r#"chore: update itertools to version 0.14.0
+
+Summary: Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod
+tempor incididunt ut labore et dolore magna aliqua.
+
+Change-Id: I1234567890abcdef1234567890abcdef12345678"#;
+
+        let trailer = parse_description_trailers(body);
+
+        // should only have Change-Id
+        assert_eq!(trailer.len(), 1);
+        assert_eq!(trailer[0].key, "Change-Id");
+    }
+
+    #[test]
+    fn test_multiline_trailer() {
+        let body = r#"chore: update itertools to version 0.14.0
+
+key: This is a very long value, with spaces and
+  newlines in it."#;
+
+        let trailer = parse_description_trailers(body);
+
+        // should only have Change-Id
+        assert_eq!(trailer.len(), 1);
+        assert_eq!(trailer[0].key, "key");
+        assert!(trailer[0].value.starts_with("This is"));
+        assert!(trailer[0].value.ends_with("in it."));
```

Comment:
nit: test the whole value so we check what happens with the newline and the leading spaces on the second line